sound/soc/sof/Kconfig

```diff
@@ -214,6 +216,7 @@ config SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST_NUM
 
 config SND_SOC_SOF_DEBUG_IPC_MSG_INJECTOR
 	tristate "SOF enable IPC message injector"
+	depends on SND_SOC_SOF
```

Comment:
Both the floodtest and injector is under `SND_SOC_SOF_DEBUG` which is under `SND_SOC_SOF_DEVELOPER_SUPPORT` and the later is only available if `SND_SOC_SOF` is enabled.

Reply:
The if statements do not force y/m selection correct. One can easily set SND_SOC_SOF_DEBUG_IPC_MSG_INJECTOR=y and SND_SOC_SOF=m, after which build fails.

Reply:
Here is an example of a config produced like that:
[broken-config.txt](https://github.com/thesofproject/linux/files/8444740/broken-config.txt)

Reply:
I see now. For consistency let's keep it like this. All config which selects `SND_SOC_SOF_CLIENT` should depend on `SND_SOC_SOF` to make sure that they will follow the core's n/m/y